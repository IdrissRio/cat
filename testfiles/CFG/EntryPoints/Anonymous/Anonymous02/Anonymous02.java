class Polygon {
  Polygon(String x) {}
  public void display() { System.out.println("Inside the Polygon class"); }
}

class Anonymous02 {
  public void createClass() {
    Polygon p1 = new Polygon("test") {
      { int x = 0; }
      public void display() {
        System.out.println("Inside an anonymous class.");
      }
      { int y = 0; }
    };
    p1.display();
  }
}
