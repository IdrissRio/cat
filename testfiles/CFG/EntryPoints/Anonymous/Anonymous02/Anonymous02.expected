NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[28] Successors: [SuperConstructorAccess[40]]
NodeName: SuperConstructorAccess[40] Successors: [Implicit[28]]
NodeName: Implicit[28] Successors: [Exit[28]]
NodeName: Exit[28] Successors: []
NodeName: Entry[40] Successors: [TypeAccess[53]]
NodeName: TypeAccess[53] Successors: [VarAccess[54]]
NodeName: VarAccess[54] Successors: [StringLiteral[58]]
NodeName: StringLiteral[58] Successors: [MethodAccess[55]]
NodeName: MethodAccess[55] Successors: [Exit[48]]
NodeName: Exit[48] Successors: []
NodeName: Entry[71] Successors: [Exit[72]]
NodeName: Exit[72] Successors: []
NodeName: Entry[63] Successors: [Exit[64]]
NodeName: Exit[64] Successors: []
NodeName: Entry[84] Successors: [StringLiteral[105]]
NodeName: StringLiteral[105] Successors: [ClassInstanceExpr[102]]
NodeName: ClassInstanceExpr[102] Successors: [VariableDeclarator[99]]
NodeName: VariableDeclarator[99] Successors: [VarAccess[177]]
NodeName: VarAccess[177] Successors: [MethodAccess[178]]
NodeName: MethodAccess[178] Successors: [Exit[92]]
NodeName: Exit[92] Successors: []
NodeName: Entry[117] Successors: [Exit[118]]
NodeName: Exit[118] Successors: []
NodeName: Entry[109] Successors: [IntegerLiteral[138]]
NodeName: IntegerLiteral[138] Successors: [VariableDeclarator[135]]
NodeName: VariableDeclarator[135] Successors: [IntegerLiteral[172]]
NodeName: IntegerLiteral[172] Successors: [VariableDeclarator[169]]
NodeName: VariableDeclarator[169] Successors: [Exit[110]]
NodeName: Exit[110] Successors: []
NodeName: Entry[140] Successors: [TypeAccess[153]]
NodeName: TypeAccess[153] Successors: [VarAccess[154]]
NodeName: VarAccess[154] Successors: [StringLiteral[158]]
NodeName: StringLiteral[158] Successors: [MethodAccess[155]]
NodeName: MethodAccess[155] Successors: [Exit[148]]
NodeName: Exit[148] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[28] Predecessor: [Implicit[28]]
NodeName: Implicit[28] Predecessor: [SuperConstructorAccess[40]]
NodeName: SuperConstructorAccess[40] Predecessor: [Entry[28]]
NodeName: Entry[28] Predecessor: []
NodeName: Exit[48] Predecessor: [MethodAccess[55]]
NodeName: MethodAccess[55] Predecessor: [StringLiteral[58]]
NodeName: StringLiteral[58] Predecessor: [VarAccess[54]]
NodeName: VarAccess[54] Predecessor: [TypeAccess[53]]
NodeName: TypeAccess[53] Predecessor: [Entry[40]]
NodeName: Entry[40] Predecessor: []
NodeName: Exit[72] Predecessor: [Entry[71]]
NodeName: Entry[71] Predecessor: []
NodeName: Exit[64] Predecessor: [Entry[63]]
NodeName: Entry[63] Predecessor: []
NodeName: Exit[92] Predecessor: [MethodAccess[178]]
NodeName: MethodAccess[178] Predecessor: [VarAccess[177]]
NodeName: VarAccess[177] Predecessor: [VariableDeclarator[99]]
NodeName: VariableDeclarator[99] Predecessor: [ClassInstanceExpr[102]]
NodeName: ClassInstanceExpr[102] Predecessor: [StringLiteral[105]]
NodeName: StringLiteral[105] Predecessor: [Entry[84]]
NodeName: Entry[84] Predecessor: []
NodeName: Exit[118] Predecessor: [Entry[117]]
NodeName: Entry[117] Predecessor: []
NodeName: Exit[110] Predecessor: [VariableDeclarator[169]]
NodeName: VariableDeclarator[169] Predecessor: [IntegerLiteral[172]]
NodeName: IntegerLiteral[172] Predecessor: [VariableDeclarator[135]]
NodeName: VariableDeclarator[135] Predecessor: [IntegerLiteral[138]]
NodeName: IntegerLiteral[138] Predecessor: [Entry[109]]
NodeName: Entry[109] Predecessor: []
NodeName: Exit[148] Predecessor: [MethodAccess[155]]
NodeName: MethodAccess[155] Predecessor: [StringLiteral[158]]
NodeName: StringLiteral[158] Predecessor: [VarAccess[154]]
NodeName: VarAccess[154] Predecessor: [TypeAccess[153]]
NodeName: TypeAccess[153] Predecessor: [Entry[140]]
NodeName: Entry[140] Predecessor: []
